infrastructure/comms/src/lib.rs

```diff
@@ -1 +1,10 @@
+#![feature(custom_attribute)]
+extern crate crypto;
```

Comment:
Accept the cargo toml changes, then you dont have to say extern crate here
currently you are running edition 2015, we should be running 2018